docs/SPIR-V.rst

```diff
@@ -0,0 +1,194 @@
+
+An example is specifying the layout of Vulkan resources::
+
+  [[using Vulkan: set(X), binding(Y)]]
+  tbuffer TbufOne {
+    [[using Vulkan: offset(Z)]]
+    float4 field;
+  };
+
+  [[using Vulkan: push_constant]]
+  tbuffer TbufTwo {
+    float4 field;
+  };
+
+  [[using Vulkan: constant_id(M)]]
+  const int specConst = N;
+
+Types
+-----
+
+Normal scalar types
++++++++++++++++++++
+
+`Normal scalar types <https://msdn.microsoft.com/en-us/library/windows/desktop/bb509646(v=vs.85).aspx>`_ in HLSL are relatively easy to handle and can be mapped directly to SPIR-V instructions:
+
+================== ==================
+      HLSL               SPIR-V
+================== ==================
+``bool``           ``OpTypeBool``
+``int``            ``OpTypeInt 32 1``
+``uint``/``dword`` ``OpTypeInt 32 0``
+``half``           ``OpTypeFloat 16``
+``float``          ``OpTypeFloat 32``
+``double``         ``OpTypeFloat 64``
+================== ==================
+
+Minimal precision scalar types
+++++++++++++++++++++++++++++++
+
+HLSL also supports various `minimal precision scalar types <https://msdn.microsoft.com/en-us/library/windows/desktop/bb509646(v=vs.85).aspx>`_, which graphics drivers can implement by using any precision greater than or equal to their specified bit precision. 
+
+- ``min16float`` - minimum 16-bit floating point value
+- ``min10float`` - minimum 10-bit floating point value
+- ``min16int`` - minimum 16-bit signed integer
+- ``min12int`` - minimum 12-bit signed integer
+- ``min16uint`` - minimum 16-bit unsigned integer
+
+There are no direct mapping in SPIR-V for these types. We may need to use ``OpTypeFloat``/``OpTypeInt`` with ``RelaxedPrecision`` for some of them and issue warnings/errors for the rest.
+
+Vectors and matrixes
+++++++++++++++++++++
+
+`Vectors <https://msdn.microsoft.com/en-us/library/windows/desktop/bb509707(v=vs.85).aspx>`_ and `matrixes <https://msdn.microsoft.com/en-us/library/windows/desktop/bb509623(v=vs.85).aspx>`_ are also relatively straightforward to handle:
+
++------------------------------------+------------------------------------+
+|               HLSL                 |             SPIR-V                 |
++------------------------------------+------------------------------------+
+|``vector<|type|, |count|``          | ``OpTypeVector |type| |count|``    |
++------------------------------------+------------------------------------+
+|``matrix<|type|, |row|, |column|>`` | ``%v = OpTypeVector |type| |row|`` |
++------------------------------------+                                    |
+|``|type||row|x|column|``            | ``OpTypeMatrix %v |column|``       |
++------------------------------------+------------------------------------+
+
+Logistics
+=========
+
+Project planning
+----------------
+
+We use `GitHub Project feature in the google fork repo <https://github.com/google/DirectXShaderCompiler/projects/1>`_ to manage tasks and track progress.
```

Comment:
Recommend capitalizing "Google" when you mean the organization.  So here, and later.